### PR TITLE
[test] fix racy test in pkg/plugins

### PR DESCRIPTION
**Pull request by AkihiroSuda**
**\- What I did**
Fixed a race in `pkg/plugins`

```
==================
WARNING: DATA RACE
Write at 0x00c42009b22e by goroutine 7:
  github.com/docker/docker/pkg/plugins.TestHttpTimeout()
      /go/src/github.com/docker/docker/pkg/plugins/client_test.go:48 +0x269
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:610 +0xc9

Previous read at 0x00c42009b22e by goroutine 12:
  github.com/docker/docker/pkg/plugins.TestHttpTimeout.func1()
      /go/src/github.com/docker/docker/pkg/plugins/client_test.go:40 +0x49
  net/http.HandlerFunc.ServeHTTP()
      /usr/local/go/src/net/http/server.go:1726 +0x51
  net/http.(*ServeMux).ServeHTTP()
      /usr/local/go/src/net/http/server.go:2022 +0xa1
  net/http.serverHandler.ServeHTTP()
      /usr/local/go/src/net/http/server.go:2202 +0xbb
  net/http.(*conn).serve()
      /usr/local/go/src/net/http/server.go:1579 +0x5f6

Goroutine 7 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:646 +0x52f
  testing.RunTests.func1()
      /usr/local/go/src/testing/testing.go:793 +0xb9
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:610 +0xc9
  testing.RunTests()
      /usr/local/go/src/testing/testing.go:799 +0x4ba
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:743 +0x12f
  main.main()
      github.com/docker/docker/pkg/plugins/_test/_testmain.go:126 +0x2f8

Goroutine 12 (running) created at:
  net/http.(*Server).Serve()
      /usr/local/go/src/net/http/server.go:2293 +0x540
  net/http/httptest.(*Server).goServe.func1()
      /usr/local/go/src/net/http/httptest/server.go:235 +0xa2
==================
```

**\- How I did it**

**\- How to verify it**
`TESTFLAGS="-race" TESTDIRS=pkg/plugins make test-unit`

**\- Description for the changelog**

**\- A picture of a cute animal (not mandatory but encouraged)**

Signed-off-by: Akihiro Suda suda.akihiro@lab.ntt.co.jp

### CODE REVIEW

**Comment by tonistiigi** [comment]
So this test takes 32 seconds? Not directly related to this change but we should definitely fix that as well.

**Comment by AkihiroSuda** [comment]
@tonistiigi makes sense, updated to take 5 seconds instead of 32.

File: pkg/plugins/client_test.go [hunk 1]
```diff
@@ -31,24 +31,30 @@ func teardownRemotePluginServer() {
 	}
 }
 
-func TestHttpTimeout(t *testing.T) {
+func testHttpTimeout(t *testing.T, timeout, epsilon time.Duration) {
```
Comment (aaronlehmann) [inline pkg/plugins/client_test.go]: Unrelated to the race fix, but I think this should be called `testHTTPTimeout`. For example, the `net/http` package has methods called `ServeHTTP`, not `ServeHttp`.

File: pkg/plugins/client_test.go [hunk 2]
```diff
@@ -31,24 +31,30 @@ func teardownRemotePluginServer() {
...
 		t.Fatalf("The request should be canceled %v", err)
 	}
+	elapsed := time.Now().Sub(begin)
+	if elapsed < timeout || elapsed > timeout+epsilon {
+		t.Fatalf("elapsed time: got %v, expected %v (epsilon=%v)",
+			elapsed, timeout, epsilon)
+	}
+}
+
+func TestHttpTimeout(t *testing.T) {
```
Comment (aaronlehmann) [inline pkg/plugins/client_test.go]: `TestHTTPTimeout`

**Comment by aaronlehmann** [comment]
Apparently golint agrees with me

```
11:43:06 ---> Making bundle: validate-lint (in bundles/1.13.0-dev/validate-lint)
11:43:06 Errors from golint:
11:43:06 pkg/plugins/client_test.go:34:6: func testHttpTimeout should be testHTTPTimeout
```

**Comment by AkihiroSuda** [comment]
thank you, updated

**Comment by aaronlehmann** [comment]
LGTM

**Comment by tonistiigi** [comment]
LGTM

**Pull request merged**